devel/nlohmann-json/Portfile

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem                      1.0
+
+PortGroup                       github 1.0
+PortGroup                       cmake 1.1
+PortGroup                       cxx11 1.1
+
+github.setup                    nlohmann json 3.2.0 v
+name                            nlohmann-json
+categories                      devel
+platforms                       darwin
+supported_archs                 noarch
+license                         MIT
+maintainers                     {gmail.com:g.litenstein @Lord-Kamina} openmaintainer
+description                     Nlohmann JSON for modern C++.
+long_description                Nlohmann is a header-only (and optionally single-header) JSON library \
+                                written in vanilla C++11. Has Unicode support\; DOM and SAX parsers, among other features.
+checksums                       rmd160  3c04b206efdeaacf49f374b33eec5b86978eef6d \
+                                sha256  2131e11d0774005e68432c5c01ff0a595e6abf0ba7979738a2017ee8a69d4001 \
+                                size    114561095
+
+configure.pre_args-append       -DBUILD_TESTING=OFF -DJSON_BuildTests=OFF
+configure.pre_args-delete       -DCMAKE_POLICY_DEFAULT_CMP0025=NEW
+configure.optflags-delete       -Os
+
+pre-test {
+    configure.args-append   -DBUILD_TESTING=ON -DJSON_BuildTests=ON
+    portconfigure::configure_main
+    portbuild::build_main
+}
```

Comment:
I see what you're doing here. It's clumsy to me though. @ryandesign may find it OK, I suppose. I would be surprised, though.

Reply:
The way @kencu suggested is what we do in other ports. It has the advantage of being "what we do", but the disadvantage of the user needing to realize that the test variant exists and selecting it if they want to run the tests.

@Lord-Kamina I could offer two objections to your method: 1) it uses `portconfigure::configure_main` and `portbuild::build_main` which are internal functions not intended for use in Portfiles. However, I've used them in Portfiles and portgroups myself, so I'd have a difficult time defending that objection. 2) It configures once in the configure phase, builds once in the build phase, and then configures and builds a second time in the test phase, which wastes the time spent during the first configure and build. But on my system the first configure takes < 2 seconds, the first build takes zero seconds because there is nothing to build since it's only headers, and the second configure takes zero seconds. So I can't object on the grounds of wasted time either.

Reply:
It's evidently just my opinion but I'd say that in cases where nothing breaks and only a pretty trivial overhead is being added, I'd rather have a hacky solution than an inconsistent interface.